src/core/font_substitutions.js

```diff
@@ -0,0 +1,458 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { validateFontName } from "./core_utils.js";
+
+const NORMAL = {
+  style: "normal",
+  weight: "normal",
+};
+const BOLD = {
+  style: "normal",
+  weight: "bold",
+};
+const ITALIC = {
+  style: "italic",
+  weight: "normal",
+};
+const BOLDITALIC = {
+  style: "italic",
+  weight: "bold",
+};
+
+const substitutionMap = new Map([
+  [
+    "Times-Roman",
+    {
+      local: [
+        "Times New Roman",
+        "Times-Roman",
+        "Times",
+        "Liberation Serif",
+        "Nimbus Roman",
+        "Nimbus Roman L",
+        "Tinos",
+        "Thorndale",
+        "TeX Gyre Termes",
+        "FreeSerif",
+        "DejaVu Serif",
+        "Bitstream Vera Serif",
+        "Ubuntu",
+      ],
+      style: NORMAL,
+      ultimate: "serif",
+    },
+  ],
+  [
+    "Times-Bold",
+    {
+      local: {
+        alias: "Times-Roman",
+        append: "Bold",
+      },
+      style: BOLD,
+      ultimate: "serif",
+    },
+  ],
+  [
+    "Times-Italic",
+    {
+      local: {
+        alias: "Times-Roman",
+        append: "Italic",
+      },
+      style: ITALIC,
+      ultimate: "serif",
+    },
+  ],
+  [
+    "Times-BoldItalic",
+    {
+      local: {
+        alias: "Times-Roman",
+        append: "Bold Italic",
+      },
+      style: BOLDITALIC,
+      ultimate: "serif",
+    },
+  ],
+  [
+    "Helvetica",
+    {
+      local: [
+        "Helvetica",
+        "Helvetica Neue",
+        "Arial",
+        "Arial Nova",
+        "Liberation Sans",
+        "Arimo",
+        "Nimbus Sans",
+        "Nimbus Sans L",
+        "A030",
+        "TeX Gyre Heros",
+        "FreeSans",
+        "DejaVu Sans",
+        "Albany",
+        "Bitstream Vera Sans",
+        "Arial Unicode MS",
+        "Microsoft Sans Serif",
+        "Apple Symbols",
+        "Cantarell",
+      ],
+      path: "LiberationSans-Regular.ttf",
+      style: NORMAL,
+      ultimate: "sans-serif",
+    },
+  ],
+  [
+    "Helvetica-Bold",
+    {
+      local: {
+        alias: "Helvetica",
+        append: "Bold",
+      },
+      path: "LiberationSans-Bold.ttf",
+      style: BOLD,
+      ultimate: "sans-serif",
+    },
+  ],
+  [
+    "Helvetica-Oblique",
+    {
+      local: {
+        alias: "Helvetica",
+        append: "Italic",
+      },
+      path: "LiberationSans-Italic.ttf",
+      style: ITALIC,
+      ultimate: "sans-serif",
+    },
+  ],
+  [
+    "Helvetica-BoldOblique",
+    {
+      local: {
+        alias: "Helvetica",
+        append: "Bold Italic",
+      },
+      path: "LiberationSans-BoldItalic.ttf",
+      style: ITALIC,
```

Comment:
This probably explains a test-failure, since it should be:
```suggestion
      style: BOLDITALIC,
```